lib/srv/app/cloud.go

```diff
@@ -191,7 +191,7 @@ func (c *cloud) getAWSSigninToken(ctx context.Context, req *AWSSigninRequest, en
 	options = append(options, func(creds *stscreds.AssumeRoleProvider) {
 		// Setting role session name to Teleport username will allow to
 		// associate CloudTrail events with the Teleport user.
-		creds.RoleSessionName = req.Identity.Username
+		creds.RoleSessionName = awsutils.TruncateRoleSessionName(req.Identity.Username)
```

Comment:
This is now truncated, which means that theoretically we can have "RoleSessionName" clashes if some names share the same 64-chars prefix.

Is this acceptable?

Should we mention that in the comment above?

Reply:
> This is now truncated, which means that theoretically we can have "RoleSessionName" clashes if some names share the same 64-chars prefix.

Theoretically yes. It's not common to have names over 64 chars. And even when that happens, they usually share the suffix, not the prefix. But agreed theoretically yes.

> Is this acceptable?

The main use for `RoleSessionName` is for Cloud Trail searching. Initially, I thought collisions are acceptable as Teleport also has audit logs for AWS access. Now thinking about it again, we can only record apis from `tsh aws`. We cannot record AWS Web Console so whatever users do there must be traced by Cloud Trail. I guess you can still reference our session start event against Cloud Trial time but it's unreliable (and cumbersome).

Alternatively, we can hash the username and use the hashed name for `RoleSessionName`. Then put the hashed name in the audit events for cross reference. Something similar to this:
https://github.com/gravitational/teleport/blob/7906e139522b26e5c75a9a2573b9065bd563208c/lib/srv/db/mysql/autousers.go#L428-L439
Well, theoretically hash can collide too =p

Reply:
I thought in similar lines to you - if the username is too long we could keep some of the prefix, hash the entire username and make the last N character the hash.

Up to you, I just wanted to make sure we had considered this.

> And even when that happens, they usually share the suffix, not the prefix

Makes sense. That makes it a bit better.

Reply:
@smallinsky what is your opinion on this? thanks 🙏

Reply:
Agree, 

We can hash whole username but this make sessionName unfriendly for the reader. 

So I agree that test beast solution for the reader will be keep prefix add a sufix as username hash: 

Let's say that we have a following username 
`verylongcomplictedusernameoftheuserthatusesteleportthatexeeds64charaters@google.com`

so we can trim username like: 

`tp-altered-verylongcompl-61521eabd63c11160345c869abba1c53e59de79`

or just skip prefix: `verylongcomplicteduserna-61521eabd63c11160345c869abba1c53e59de79` 

that will be a bit more valuable for the reader instead of just random sha1 hash.

Reply:
> verylongcomplicateduserna-61521eabd63c11160345c869abba1c53e59de79

That's perfect. I will add a `role_session_name` field in audit events and save this value.

@joaoubaldo, as discussed, I will take it from here and update the changes in this PR.

Reply:
I like that approach.
If it exceeds the max, then replace the last 8 or 16 chars with some hash of the full string.

Reply:
Updated. Decide to keep the entire 40byte of sha1.